File: Practica12/EJ3/Java/Estrategia.java
interface Estrategia {
    void ejecutar();
}
class EstrategiaA implements Estrategia {
    public void ejecutar() {
        System.out.println("Estrategia A ejecutada.");
    }
}
class EstrategiaB implements Estrategia {
    public void ejecutar() {
        System.out.println("Estrategia B ejecutada.");
    }
}
class Contexto {
    private Estrategia estrategia;
    public void setEstrategia(Estrategia estrategia) {
        this.estrategia = estrategia;
    }
    public void ejecutarEstrategia() {
        estrategia.ejecutar();
    }
    public static void main(String[] args) {
        Contexto c = new Contexto();
        c.setEstrategia(new EstrategiaA());
        c.ejecutarEstrategia();
        c.setEstrategia(new EstrategiaB());
        c.ejecutarEstrategia();
    }
}

File: Practica12/EJ3/Python/Estrategia.py
class Estrategia:
    def ejecutar(self):
        pass
class EstrategiaA(Estrategia):
    def ejecutar(self):
        print("Estrategia A ejecutada.")
class EstrategiaB(Estrategia):
    def ejecutar(self):
        print("Estrategia B ejecutada.")
class Contexto:
    def __init__(self, estrategia: Estrategia):
        self.estrategia = estrategia
    def setEstrategia(self, estrategia: Estrategia):
        self.estrategia = estrategia
    def ejecuta(self):
        self.estrategia.ejecutar()
c = Contexto(EstrategiaA())
c.ejecuta()
c.setEstrategia(EstrategiaB())
c.ejecuta()
